File: 1.py
def iStepFibbonacci(n):
    f = [0, 1]
    for i in range(2, n + 1):
        f.append(f[i - 1] + f[i - 2])
    return f[n]
def rStepFibbonacci(n):
    if n < 0:
        return 0
    elif n == 0 or n == 1:
        return 1
    else:
        return rStepFibbonacci(n - 1) + rStepFibbonacci(n - 2)
def main():
    n = int(input("Enter NO: "))
    fibI = iStepFibbonacci(n)
    fibR = rStepFibbonacci(n)
    print(f"Fibonacci Value (Iteration): {fibI}")
    print(f"Fibonacci Value (Recursion): {fibR}")
    print(f"Steps required using Iteration: {n}")
    print(f"Steps required using Recursion: {rStepFibbonacci(n)}")
if __name__ == "__main__":
    main()

File: 1.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> f; // Declare the Fibonacci sequence vector globally

int iStepFibbonacci(int n) {
    f.push_back(0);
    f.push_back(1);
    
    for (int i = 2; i <= n; i++) {
        f.push_back(f[i - 1] + f[i - 2]);
    }
    
    return f[n]; // Return the nth Fibonacci number, not 'n'
}

int rSteps = 0; // Count recursive steps

int rStepFibbonacci(int n) {
    rSteps++;
    if (n < 0)
        return 0;
    if (n == 1 || n == 0)
        return 1;
    return rStepFibbonacci(n - 1) + rStepFibbonacci(n - 2);
}

int main() {
    int n;
    cout << "Enter NO: ";
    cin >> n;

    // Clear the global vector before each run
    f.clear();

    // Calculate Fibonacci values and steps
    int fibI = iStepFibbonacci(n);
    int fibR = rStepFibbonacci(n);

    cout << "Fibonacci Value (Iteration): " << fibI << '\n';
    cout << "Fibonacci Value (Recursion): " << fibR << '\n';
    cout << "Steps required using Iteration: " << n << '\n';
    cout << "Steps required using Recursion: " << rSteps << '\n';

    return 0;
}
